lib/srv/app/server.go

```diff
@@ -465,6 +465,17 @@ func (s *Server) close(ctx context.Context) error {
 	g, gctx := errgroup.WithContext(ctx)
 	g.SetLimit(100)
 
+	select {
+	case sender := <-s.c.InventoryHandle.Sender():
+		// Manual deletion per app is only required if the auth server
+		// doesn't support actively cleaning up app resources when the
+		// inventory control stream is terminated during shutdown.
+		if capabilities := sender.Hello().Capabilities; capabilities != nil {
+			shouldDeleteApps = shouldDeleteApps && !capabilities.AppCleanup
+		}
+	default:
+	}
```

Comment:
non-blocking selects on `Sender()` channel are pretty racy.  It'll fail if either a concurrent goroutine is also pulling the sender, or if a downstream message arrives concurrently.  This should either be a blocking select with a timeout, or we should add some kind of official mechanism of doing non-blocking sender acquisition (e.g. `GetSender() (s DownstreamSender, ok bool)`.